src/clickhouse/mod.rs

```diff
@@ -0,0 +1,137 @@
+use std::collections::BTreeMap;
+
+use testcontainers::{core::WaitFor, Image, ImageArgs};
+
+const DEFAULT_USER: &str = "default";
+const DEFAULT_DATABASE: &str = "clickhouse";
+
+const DEFAULT_IMAGE_NAME: &str = "clickhouse/clickhouse-server";
+const DEFAULT_IMAGE_TAG: &str = "23.3.8.21-alpine";
+
+/// Module to work with [`ClickHouse`] inside of tests.
+///
+/// This module is based on the official [`ClickHouse docker image`].
+///
+/// # Example
+/// ```
+/// use testcontainers_modules::{clickhouse, testcontainers::runners::SyncRunner};
+///
+/// let clickhouse = clickhouse::ClickHouse::default().start().unwrap();
+/// let http_port = clickhouse.get_host_port_ipv4(26257).unwrap();
+///
+/// // do something with the started clickhouse instance..
+/// ```
+///
+/// [`ClickHouse`]: https://clickhouse.com/
+/// [`Clickhouse docker image`]: https://hub.docker.com/r/clickhouse/clickhouse-server
+#[derive(Debug)]
+pub struct ClickHouse {
+    name: String,
+    tag: String,
+    env_vars: BTreeMap<String, String>,
+}
+
+impl Default for ClickHouse {
+    fn default() -> Self {
+        ClickHouse::new(
+            DEFAULT_IMAGE_NAME.to_string(),
+            DEFAULT_IMAGE_TAG.to_string(),
+        )
+    }
+}
+
+impl ClickHouse {
+    fn new(name: String, tag: String) -> Self {
+        ClickHouse {
+            name,
+            tag,
+            env_vars: Default::default(),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ClickHouseArgs {
+    command: String,
+    args: Vec<String>,
+}
+
+impl Default for ClickHouseArgs {
+    fn default() -> Self {
+        Self {
+            command: "start-single-node".to_string(),
+            args: vec!["--insecure".to_string()],
+        }
+    }
+}
```

Comment:
I think we should expose builder methods for these arguments.

Command can be hardcoded for now, but I think it's important to allow to override `insecure` at least

Reply:
This was copy pasta 🍝